CMakeLists.txt

```diff
@@ -633,23 +583,138 @@ if (WIN32)
+		DESCRIPTION "enable deprecated functions for backwards compatibility" DISABLED TORRENT_NO_DEPRECATE)
 endif()
+feature_option(encryption "Enables encryption in libtorrent" ON)
+feature_option(exceptions "build with exception support" ON)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME extensions DEFAULT ON
+	DESCRIPTION "Enables protocol extensions" DISABLED TORRENT_DISABLE_EXTENSIONS
+)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME extensions DEFAULT ON
+	DESCRIPTION "Enables protocol extensions" DISABLED TORRENT_DISABLE_EXTENSIONS)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME i2p DEFAULT ON
+	DESCRIPTION "build with I2P support" DISABLED TORRENT_USE_I2P=0)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME ipv6 DEFAULT ON
+	DESCRIPTION "build with IPv6 support" DISABLED TORRENT_USE_IPV6=0)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME logging DEFAULT ON
+	DESCRIPTION "build with logging" DISABLED TORRENT_DISABLE_LOGGING)
+
+find_public_dependency(Iconv)
+if(MSVC)
+	set(iconv_package_type OPTIONAL)
+else()
+	set(iconv_package_type RECOMMENDED)
+endif()
+
+set_package_properties(Iconv
+	PROPERTIES
+		URL "https://www.gnu.org/software/libiconv/"
+		DESCRIPTION "GNU encoding conversion library"
+		TYPE ${iconv_package_type}
+		PURPOSE "Convert strings between various encodings"
+)
+
+if(Iconv_FOUND)
+	target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_ICONV)
+	target_include_directories(torrent-rasterbar PRIVATE ${ICONV_INCLUDE_DIR})
+	target_link_libraries(torrent-rasterbar PRIVATE ${ICONV_LIBRARIES})
+endif()
+
+find_public_dependency(OpenSSL)
+set_package_properties(OpenSSL
+	PROPERTIES
+		URL "https://www.openssl.org/"
+		DESCRIPTION "Full-strength general purpose cryptography library"
+		TYPE RECOMMENDED
+		PURPOSE "Provides HTTPS support to libtorrent"
+)
+
+if(OPENSSL_FOUND)
+	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
+	target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_OPENSSL)
+	target_sources(torrent-rasterbar PRIVATE src/pe_crypto)
+endif()
+
+if (encryption)
+	if (OPENSSL_FOUND)
+		target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_LIBCRYPTO)
+	else(OPENSSL_FOUND)
+		find_public_dependency(LibGcrypt)
+		set_package_properties(LibGcrypt
+			PROPERTIES
+				URL "https://www.gnupg.org/software/libgcrypt/index.html"
+				DESCRIPTION "A general purpose cryptographic library"
+				TYPE RECOMMENDED
+				PURPOSE "Use GCrypt instead of the built-in functions for RC4 and SHA1"
+		)
+		if (LibGcrypt_FOUND)
+			target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_LIBGCRYPT)
+			target_link_libraries(torrent-rasterbar PRIVATE LibGcrypt::LibGcrypt)
+		else()
+			if (NOT WIN32 AND NOT APPLE)
+				target_sources(torrent-rasterbar PRIVATE src/sha1)
```

Comment:
sha1.cpp is needed even if encryption is disabled as long as there is no crypto library found and we're not building for WIN32 or APPLE.

Reply:
Thanks! Carefully copied features from the Jamfile this time.